packages/medusa/src/types/pricing.ts

```diff
@@ -31,6 +31,35 @@ export type ShippingOptionPricing = {
   tax_amount: number
 }
 
+/** @schema PricedShippingOption
+ * title: "Priced Shipping Option"
+ * type: object
+ * allOf:
+ *   - $ref: "#/components/schemas/ShippingOption"
+ *   - type: object
+ *     properties:
+ *       price_incl_tax:
+ *         type: number
+ *         description: Price including taxes
+ *       tax_rates:
+ *         type: array
+ *         description: An array of applied tax rates
+ *         items:
+ *           type: object
+ *           properties:
+ *             rate:
+ *               type: number
+ *               description: The tax rate value
+ *             name:
+ *               type: string
+ *               description: The name of the tax rate
+ *             code:
+ *               type: string
+ *               description: The code of the tax rate
+ *       tax_amount:
+ *         type: number
+ *         description: The taxes applied.
```

Comment:
suggestion: should we specify anything as required here?

Reply:
I don't believe so. I followed the same pattern as PriceVariant.